packages/salesforcedx-utils-vscode/src/helpers/paths.ts

```diff
@@ -137,6 +138,22 @@ function sfdxProjectConfig(): string {
   return pathToSFDXProjectConfig;
 }
 
+function sfdxGlobalConfig(): string {
+  const pathToSFDXGlobalConfig = path.join(
+    os.homedir(),
+    relativeStateFolder(),
+    SFDX_CONFIG_FILE
+  );
+  return pathToSFDXGlobalConfig;
+}
+
+function sfdxConfig(): string {
```

Comment:
So quick question When would we ever want to get a config file and not know if it's global or project level? I'll review the use case in this PR but that seems like a dangerous flow.

Reply:
The way this was originally written, I thought maybe the path to the global config file was desired when a VS Code folder was open that wasn't an sfdx project.  But, it doesn't make a lot of sense to look for the global sfdx config file if you are not in an sfdx project at all.  When I fire up the extensions when NOT in an sfdx project, this code isn't hit at all since this code block is not run:
<img width="400" alt="2022-10-28_14-19-54" src="https://user-images.githubusercontent.com/46458081/198734737-acc97d27-818f-4ef9-ae58-29ec2becc10c.png">
So, maybe the project level fn (`sfdxProjectConfig()`) is the only one that is actually needed

Reply:
yeah that's my suspicion.  Looks like you already did the deeper digging work I was gonna do.  I'd say we get rid of that and just only use the workspace config.  

So the only commands that can be run when not in a sf project are listed in the activation events ```    "workspaceContains:sfdx-project.json",
    "onCommand:sfdx.force.project.create",
    "onCommand:sfdx.force.project.with.manifest.create",
    "onCommand:sfdx.debug.isv.bootstrap"```

Project create and create with manifest do not depend on have a default username...check on isv.bootstrap now

Reply:
Verified...there is a check already in that sfdx.debug.isv.bootstrap command that we are in a sfdx project so we can get rid of the global config file watch 👍

Reply:
hum....so I can never leave well enough alone.  I looked at the commit where the global was added and it was specifically around project create.  So if we are gonna switch that to only respect local we should def verify the empty state -> create project command still works as expected.  I would think so, but best to be sure.

Reply:
What would you think about merging this, which maintains previous behavior, and follow up WIs to 
A: investigate whether the global path can be removed from this flow, and 
B: to refactor what this flow actually does (scratch org decorator reads the config file and can use ConfigUtil instead)
<img width="400" alt="2022-10-28_15-09-10" src="https://user-images.githubusercontent.com/46458081/198742969-a0b36818-4c47-4abc-97ae-bc7e8a650629.png">

Reply:
I'm fine with that. Good call on configUtils.  I would prefer that we get rid of the `sfdxConfig` function and just use workspaceUtils in the calling code ala 
```
// In an ideal world we'd log if we using local or global config file.
const configFile = workspaceUtils.hasRootWorkspace() ? sfdxProjectConfig() : sfdxGlobalConfig();
```
Getting a configFile and not knowing which one just seems like it'd be a bad thing.

Reply:
Yep.  If scratch org decorator is updated to use `ConfigUtil.getDefaultUsernameOrAlias()` inside `displayDefaultUserName`, then both `paths.sfdxConfig` and `paths.sfdxGlobalConfig` can be removed.